### PR TITLE
keras-apache-mxnet update with bug fixes and new features

### PR DESCRIPTION
**Pull request by roywei**
In this PR to master, we will include a new feature, several bug fixes, and add a missing operator.

# Changes

## New Features:  Save mxnet model during training using checkpoints

**Related PRs:**
- Add saving model checkpoint API #132 
- Add get mxnet model info API #144 
- Tutorial update #147

**Documentation:**

- Save MXNet Model tutorial #147 

## Bug Fixes 

1. Fix batchnorm bug when gamma and beta is None with PR #137 (related issues: #119, #136)
2. Fix Conv1D Layer bug with channels first format with PR #143 (related issues: #121 , #122)
3. Fix element-wise comparison operators bug with PR #145  (related issues: #146)

## Improvements

1. Fix CI to track PR changes instead of master with PR #141 (related issues: #123 )

## Missing Operators
1. Add missing operators: sparse categorical crossentropy with PR #145

### CODE REVIEW

**Comment by sandeep-krishnamurthy** [comment]
Love it!
Conv operator fixes is also going in.

**Comment by kalyc** [comment]
Thanks for opening this PR tracking the patch release @roywei! Would we be having another release to track Keras-2.2.2? Or will this patch release be a part of the new version?

File: docs/mxnet_backend/save_mxnet_model.md [hunk 1]
```diff
@@ -3,32 +3,41 @@
...
+You can now use Keras-MXNet for training the model and MXNet for inference in production. You can use `keras.models.save_mxnet_model()` API to save
+the models trained on the CPU, a single GPU or multiple GPUs or, you can use `keras.callbacks.MXNetModelCheckpoint` callback to checkpoint the model.
 
 You can use any language bindings supported by MXNet (Scala/Python/Julia/C++/R/Perl) for performing inference with these models!
 
 `Warning` Not all Keras operators and functionalities are supported with MXNet backend. For more information, view the the list
  of known issues and unsupported functionalities [here](https://github.com/awslabs/keras-apache-mxnet/issues/18).
 
+
+## Export MXNet model using save_mxnet_model() API
```
Comment (kalyc) [inline docs/mxnet_backend/save_mxnet_model.md]: shall we add links to both the scala & MMS blog posts to show how the exported model can be further used with another Deep learning utility?

**Comment by roywei** [comment]
@sandeep-krishnamurthy @kalyc  I would like to merge this PR to master. smoke tested working fine, enabled two more examples (babi_memnn.py and mnist_acgan.py)

File: docs/mxnet_backend/save_mxnet_model.md (hunk 1 above)
Comment (roywei) [reply docs/mxnet_backend/save_mxnet_model.md]: see updates in #151
